starlite/testing.py

```diff
@@ -85,6 +88,25 @@ def __init__(
             backend_options=backend_options,
         )
 
+    def __enter__(self: T_client, *args: Any, **kwargs: Any) -> T_client:
+        """
+        Starlette's `TestClient.__enter__()` return value is strongly typed to return their own
+        `TestClient`, i.e., not-generic to support subclassing.
+
+        Override here to provide a nicer typing experience downstream from us.
+
+        Parameters
+        ----------
+        args : Any
+        kwargs : Any
+            `*args, **kwargs` passed straight through to `Starlette.testing.TestClient.__enter__()`
+
+        Returns
+        -------
+        TestClient
+        """
+        return super().__enter__(*args, **kwargs)  # type:ignore[return-value]
```

Comment:
you can try ignore[override]

Reply:
I don't follow. If I change that to `# type:ignore[override]` mypy throws:

```
starlite/testing.py:108: error: Unused "type: ignore" comment
starlite/testing.py:108: error: Incompatible return value type (got "TestClient", expected "T_client")  [return-value]
starlite/testing.py:108: note: Error code "return-value" not covered by "type: ignore" comment
```

Reply:
I"m just holding on this @Goldziher, have I misunderstood?